src/adam_impact_study/physical_params.py

```diff
@@ -0,0 +1,310 @@
+    """
+    Determine the asteroid class based on the percentage of C and S asteroids.
+
+    Parameters
+    ----------
+    percent_C : float
+        Percentage of C asteroids.
+    percent_S : float
+        Percentage of S asteroids.
+
+    Returns
+    -------
+    ast_class : str
+        Asteroid class.
+    """
+    assert percent_C + percent_S == 1, "Percentage of C and S asteroids must equal 1"
+    rng = np.random.default_rng()
+    return "C" if rng.random() < percent_C else "S"
+
+
+def calculate_H(diameter: float, albedo: float) -> float:
+    """
+    Calculate the absolute magnitude of an asteroid.
+
+    Parameters
+    ----------
+    diameter : float
+        Asteroid diameter, in kilometers.
+    albedo : float
+        Asteroid albedo.
+
+    Returns
+    -------
+    H : float
+        Absolute magnitude.
+    """
+    return 15.618 - 5 * np.log10(diameter) - 2.5 * np.log10(albedo)
+
+
+def load_config(file_path: str, run_id: str = None) -> ImpactorConfig:
+    """
+    Load the impact study configuration from a file.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the configuration file.
+    run_id : str, optional
+        User-defined run ID. If not provided, the run ID will be extracted from the file name.
+
+    Returns
+    -------
+    config : `~adam_impact_study.physical_params.ImpactorConfig`
+        Configuration object.
+    """
+
+    if run_id is None:
+        run_id = os.path.basename(file_path).split(".")[0]
+
+    with open(file_path, "r") as file:
+        config_data = json.load(file)
+
+    S_type = ImpactorConfig.from_kwargs(
+        config_id=[f"S_type_{run_id}"],
+        ast_class=["S"],
+        albedo_min=[config_data.get("S_albedo_min", 0.10)],
+        albedo_max=[config_data.get("S_albedo_max", 0.22)],
+        albedo_distribution=["uniform"],
+        percentage=[config_data.get("percent_S", 0.5)],
+        min_diam=[config_data.get("min_diam", 0.01)],
+        max_diam=[config_data.get("max_diam", 1)],
+        u_r=[config_data.get("u_r_S", 2.182)],
+        g_r=[config_data.get("g_r_S", 0.65)],
+        i_r=[config_data.get("i_r_S", -0.2)],
+        z_r=[config_data.get("z_r_S", -0.146)],
+        y_r=[config_data.get("y_r_S", -0.151)],
+    )
+
+    C_type = ImpactorConfig.from_kwargs(
+        config_id=[f"C_type_{run_id}"],
+        ast_class=["C"],
+        albedo_min=[config_data.get("C_albedo_min", 0.03)],
+        albedo_max=[config_data.get("C_albedo_max", 0.09)],
+        albedo_distribution=["uniform"],
+        percentage=[config_data.get("percent_C", 0.5)],
+        min_diam=[config_data.get("min_diam", 0.01)],
+        max_diam=[config_data.get("max_diam", 1)],
+        u_r=[config_data.get("u_r_C", 1.786)],
+        g_r=[config_data.get("g_r_C", 0.474)],
+        i_r=[config_data.get("i_r_C", -0.119)],
+        z_r=[config_data.get("z_r_C", -0.126)],
+        y_r=[config_data.get("y_r_C", -0.131)],
```

Comment:
We could store the config file using ImpactorConfig, either as parquet or as csv. (e.g. to_csv). CSV would allow it to be human readable, if that's important. Then you can just load it in with a single command.